src/http.c

```diff
@@ -483,12 +508,53 @@ http_stream_setup(char **stream, const char *url)
   while ((line = evbuffer_readln(ctx.input_body, NULL, EVBUFFER_EOL_ANY)) && (n < 10))
     {
       n++;
-      if (strncasecmp(line, "http://", strlen("http://")) == 0)
+      if (is_m3u)
 	{
-	  DPRINTF(E_DBG, L_HTTP, "Found internet playlist stream (line %d): %s\n", n, line);
+          if ( strncasecmp(line, "http://", sizeof("http://")-1) == 0 )
+            {
+              DPRINTF(E_DBG, L_HTTP, "Found internet playlist (m3u) stream (line %d): %s\n", n, line);
 
-	  n = -1;
-	  break;
+              n = -1;
+              break;
+            }
+	}
+      else
+	{
+          if (!in_playlist)
```

Comment:
I think you should document what you are trying to parse in the below, e.g. by writing an example line

Reply:
Done.